packages/clerk-js/src/core/devBrowser.ts

```diff
@@ -0,0 +1,105 @@
+import { getDevBrowserJWTFromURL, setDevBrowserJWTInURL } from '@clerk/shared/devBrowser';
+
+import { isDevOrStagingUrl } from '../utils';
+import { getDevBrowserCookie, removeDevBrowserCookie, setDevBrowserCookie } from '../utils/cookies/devBrowser';
+import { clerkErrorDevInitFailed } from './errors';
+import type { FapiClient } from './fapiClient';
+
+export interface DevBrowser {
+  clear(): void;
+
+  setup(): Promise<void>;
+
+  getDevBrowserJWT(): string | undefined;
+
+  setDevBrowserJWT(jwt: string): void;
+
+  removeDevBrowserJWT(): void;
+}
+
+export type CreateDevBrowserOptions = {
+  frontendApi: string;
+  fapiClient: FapiClient;
+};
+
+export function createDevBrowser({ frontendApi, fapiClient }: CreateDevBrowserOptions): DevBrowser {
+  function getDevBrowserJWT() {
+    return getDevBrowserCookie();
+  }
+
+  function setDevBrowserJWT(jwt: string) {
+    setDevBrowserCookie(jwt);
+  }
+
+  function removeDevBrowserJWT() {
+    removeDevBrowserCookie();
+  }
+
+  function clear() {
+    removeDevBrowserJWT();
+  }
+
+  async function setup(): Promise<void> {
+    if (!isDevOrStagingUrl(frontendApi)) {
+      return;
+    }
+
+    // 1. Set network interceptors to Pass dev
+    fapiClient.onBeforeRequest(request => {
+      const devBrowserJWT = getDevBrowserJWT();
+      if (devBrowserJWT && request?.url) {
+        request.url = setDevBrowserJWTInURL(request.url, devBrowserJWT, true);
+      }
+    });
+
+    fapiClient.onAfterResponse((_, response) => {
+      const newDevBrowserJWT = response?.headers?.get('Clerk-Db-Jwt');
+      if (newDevBrowserJWT) {
+        setDevBrowserJWT(newDevBrowserJWT);
+      }
+    });
+
+    // 1. If a cookie already exists, it might have SameSite=Strict. Re-set it to make sure it has SameSite=Lax
+    const existingDevBrowserCookie = getDevBrowserCookie();
+    if (existingDevBrowserCookie) {
+      removeDevBrowserCookie();
+      setDevBrowserCookie(existingDevBrowserCookie);
+    }
+
+    // 2. Get the JWT from hash or search parameters when the redirection comes from AP
```

Comment:
Hedging considering we're going to drop the hash transport mechanism:

```suggestion
    // 2. Get the JWT from the URL when the redirection comes from AP
```

Reply:
Good call. I have a to-do item for this, but I'd like us to see how V4 and V5 will work in AP before the final change. So we will get back to this next week.